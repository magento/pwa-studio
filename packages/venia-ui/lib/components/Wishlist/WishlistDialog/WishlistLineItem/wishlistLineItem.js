import React, { useCallback } from 'react';
import { useStyle } from '@magento/venia-ui/lib/classify';

import defaultClasses from './wishlistLineItem.css';
import { bool, func, number } from 'prop-types';

const WishlistLineItem = props => {
    const { id, isDisabled, onClick } = props;
    const classes = useStyle(defaultClasses, props.classes);

    const handleClick = useCallback(() => {
        onClick(id);
    }, [id, onClick]);

    return (
        <button
            className={classes.root}
            disabled={isDisabled}
            type="button"
            onClick={handleClick}
        >
            {props.children}
        </button>
    );
};

export default WishlistLineItem;

WishlistLineItem.defaultProps = {
    id: number,
    isDisabled: bool,
    onClick: func.isRequired
};
