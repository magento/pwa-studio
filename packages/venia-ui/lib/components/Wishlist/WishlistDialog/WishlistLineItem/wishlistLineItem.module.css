.root {
    width: 100%;
    text-align: left;

    border-bottom: 1px solid rgb(var(--venia-global-color-gray-dark));
    font-weight: var(--venia-global-fontWeight-semibold);
    height: 3.5rem;
}
