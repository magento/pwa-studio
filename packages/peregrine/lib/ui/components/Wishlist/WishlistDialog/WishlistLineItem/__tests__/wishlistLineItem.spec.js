import React from 'react';
import createTestInstance from '@magento/peregrine/lib/util/createTestInstance';
import WishlistLineItem from '../wishlistLineItem';
import { act } from 'react-test-renderer';

jest.mock('@magento/venia-ui/lib/classify');

const onClickMock = jest.fn();
const defaultProps = {
    id: 1,
    isDisabled: false,
    onClick: onClickMock
};

test('renders the correct tree', () => {
    const tree = createTestInstance(<WishlistLineItem {...defaultProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders a disabled button', () => {
    const props = { ...defaultProps, disabled: true };
    const { root } = createTestInstance(<WishlistLineItem {...props} />);
    expect(root.findByProps({ className: 'root' }).props.disabled).toBe(false);
});

test('clicking invokes onClick prop with id', () => {
    const props = { ...defaultProps, disabled: true };
    const { root } = createTestInstance(<WishlistLineItem {...props} />);

    act(() => {
        root.findByProps({ className: 'root' }).props.onClick();
    });

    expect(onClickMock).toHaveBeenCalled();
});
