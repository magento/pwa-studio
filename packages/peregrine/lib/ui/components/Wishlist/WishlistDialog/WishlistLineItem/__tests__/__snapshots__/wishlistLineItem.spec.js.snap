// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct tree 1`] = `
<button
  className="root"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <h2
    className="lineItemName"
  />
</button>
`;
